pkg/apply/event/event.go

```diff
@@ -152,6 +154,7 @@ const (
 )
 
 type DeleteEvent struct {
+	GroupName  string
```

Comment:
It's necessary in order to sort the events by group. Otherwise there's no way to tell which group the apply/prune/delete event is from without keeping track of the last ActionGroupEvent. 

I guess I could modify the sort algo to go backwards in the event list until it finds an ActionGroupEvent, but that seems like a hack. It seems like a much cleaner API to put it in the event itself.

Reply:
Going backwards in the sort algo would depend on the implementation detail that Golang's sort algo is a stable in-place sort. It might work, if changing the event API is not ok, but I figured embedding the GroupName made for better event handling UX.

Reply:
Perhaps a better reason:
If we add async processing of dependency trees later, we're not going to be able to depend on only having one ActionGroup at a time. Adding GroupName now allows them to be emitted and handled asynchronously later.

Reply:
Golang has separate sorting functions if a stable sort is needed, so we can do this without relying on implementation details (the normal golang sort is not stable as far as I know). I agree that we might need this in the events eventually, but ideally I think we should wait until we really understand what we need before doing it. Is it very awkward to do this without changing the API?

Reply:
As discussed, I put the GroupName change in its own commit, since it's a minor API change.